java/.gitignore

```diff
@@ -0,0 +1,239 @@
+/.idea/caches
```

Comment:
I think this gitignore is too verbose for what it should normally be. I recommend using wildcard matching.